spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/InputWrapper.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn.mkldnn
+
+import com.intel.analytics.bigdl.nn.abstractnn.Activity
+
+private[bigdl] class InputWrapper extends MklDnnLayer {
```

Comment:
why do we need  a separate wrapper for input ? we already have BLASWrapper

Reply:
BLASWrapper is only for blas layer wrapper. And InputWrapper is only to wrap dnn input layer.
For dnn input layer, it needs  input shape (usually batch size and image size) and format to init.
So when we create a dnn model, we must have input shape and can not be changed.  It means inference must use the same batch size with training.

In order to fix this issue, I made two points change:
1. create primitives in IRgraph only when forward / backward is called.
2. Init dnn input layer in InputWrapper, not creating dnn model